tracer/src/Datadog.Trace/ClrProfiler/CallTarget/CallTargetReturn.cs

```diff
@@ -2,73 +2,73 @@
 // Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
 // This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
 // </copyright>
+#nullable enable
 
 using System.ComponentModel;
 using System.Runtime.CompilerServices;
 
-namespace Datadog.Trace.ClrProfiler.CallTarget
+namespace Datadog.Trace.ClrProfiler.CallTarget;
+
+/// <summary>
+/// Call target return value
+/// </summary>
+/// <typeparam name="T">Type of the return value</typeparam>
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public readonly ref struct CallTargetReturn<T>
 {
+    private readonly T? _returnValue;
```

Comment:
[Spent some time reading the implications of this](https://learn.microsoft.com/en-us/dotnet/csharp/nullable-references#generics):

- If the type argument for `T` is a reference type, `T?` references the corresponding nullable reference type. For example, if `T` is a `string`, then `T?` is a `string?`.
- If the type argument for `T` is a value type, `T?` references the same value type, `T`. For example, if `T` is an `int`, the `T?` is also an `int`.
- If the type argument for `T` is a nullable reference type, `T?` references that same nullable reference type. For example, if `T` is a `string?`, then `T?` is also a `string?`.
- If the type argument for `T` is a nullable value type, `T?` references that same nullable value type. For example, if `T` is a `int?`, then `T?` is also a `int?`.

So I think we're ok. The rules change _again_ if you add `class` constraints but we're ok here 😅

Reply:
great, yeah that's the right behaviour for our case.